enhancements/sig-architecture/81-addon-lifecycle/README.md

```diff
@@ -37,26 +38,787 @@ proposal's scope.
+Create a new `AddOnHubConfig` with the latest version,
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: AddOnHubConfig
+metadata:
+  name: hub-config-yyy
+spec:
+  desiredVersion: v0.11.0
+```
+
+The config name inside `ClusterManagementAddOn` will change at the same time. 
+
+The admin can create a placement `canary-placement` to select the canary upgrade clusters.
+Then append `rolloutStrategy` and refer to the `canary-placement` so that the new config will only apply to the canary clusters.
+
+The addon manager watches the change and applies the new config to the selected clusters.
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: helloworld
+spec:
+  configuration:
+    configs:
+    - group: addon.open-cluster-management.io
+      resource: addondeploymentconfigs
+      name: deploy-config-xxx
+      namespace: open-cluster-management
+    - group: addon.open-cluster-management.io
+      resource: addonhubconfigs
+      name: hub-config-yyy
+    rolloutStrategy:
+      type: RollingUpdateWithPlacement
+      rollingUpdateWithPlacement:
+        name: canary-placement
+        namespace: default
+        maxConcurrentlyUpdating: 25%
+  installStrategy:
+    type: Placements
+    placements:
+    - name: aws-placement
+      namespace: default
+      addonSpec:
+        installNamespace: open-cluster-management-agent-addon
+        configs:
+          - group: addon.open-cluster-management.io
+            resource: addondeploymentconfigs
+            name: aws-config
+    - name: gcp-placement
+      namespace: default
+      addonSpec:
+        installNamespace: open-cluster-management-agent-addon
+        configs:
+        - group: addon.open-cluster-management.io
+          resource: addondeploymentconfigs
+          name: gcp-config
+```
+
+The `ManagedClusterAddOn` `status.configReferences` updates to new config.
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddOn
+metadata:
+  name: helloworld
+  namespace: cluster1
+spec:
+  installNamespace: open-cluster-management-agent-addon
+  configs:
+    - group: addon.open-cluster-management.io
+      resource: addondeploymentconfigs
+      name: aws-config
+Status:
+…
+  supportedConfigs:
+  - group: addon.open-cluster-management.io
+    resource: addondeploymentconfigs
+  configReferences:
+  - group: addon.open-cluster-management.io
+    resource: addondeploymentconfigs
+    name: aws-config
+    namespace: open-cluster-management
+    desiredConfigSpecHash: <aws-config-spec-hash>
+    lastAppliedConfigSpecHash: <aws-config-spec-hash>
+  - group: addon.open-cluster-management.io
+    resource: addonhubconfigs
+    name: hub-config-yyy
+    desiredConfigSpecHash: <hub-config-yyy-spec-hash>
+    lastAppliedConfigSpecHash: <hub-config-xxx-spec-hash>
+  conditions:
+  - lastTransitionTime: "2022-11-02T05:48:12Z"
+    message: upgrading...
```

Comment:
We should be able to indicate how many are at sha/X, sha/Y, and sha/Z, right?

Reply:
Considering adding `installProgression` to the `ClusterManagementAddOn` status, more details should be fulfilled there.